tests/components/litterrobot/test_sensor.py

```diff
@@ -85,7 +88,7 @@ async def test_litter_robot_sensor(
     assert sensor.state == "dfs"
     assert sensor.attributes["device_class"] == "litterrobot__status_code"
     sensor = hass.states.get("sensor.test_litter_level")
-    assert sensor.state == "70.0"
+    assert sensor.state == "0.0"
```

Comment:
Using freeze_time has solved some of the issues, but I can't work out how to solve the issue with litter-level

Reply:
More investigation shows that:
- with pylitterbot==`2022.9.2` it return `66.7`
- with pylitterbot==`2022.9.3` it return `0.0`

So it looks to me like maybe there is underlying bug somewhere?

Reply:
Test data was pinned to `datetime.now()` before this PR. So I thought I prevented any flaky behavior. Thanks. Will have to check this out. The freeze time value may just have to be adjusted?

Reply:
I'll release a fix for the litter_level in the dependency that this freeze inadvertently creates and update the tests. Just to shed some light on what is happening on these various pieces:

- The sleep mode start/end time runs on a recurring schedule. Once set on the robot, it continues that cycle regardless of whether the robot is online or not. The python package mimics this behavior by updating the datetime after the previous start or end time passes. Because a sensor requires UTC timestamps rather than just a time portion, the test against it must either dynamically create the date based on today or we freeze the time so we guarantee what that time is.
- During a clean cycle, the litter level sometimes is measured mid-cycle which causes a short spike. The python package uses a datetime expiration to try to account for this, while also allowing that spike to become permanent if, say, an empty cycle was performed and that mid-cycle measurement ends up being the same as the end-cycle measurement. I naively created this expiration on instantiation with `datetime.now(timezone.utc)` (https://github.com/natekspencer/pylitterbot/blob/main/pylitterbot/robot/litterrobot4.py#L96) which is then checked when trying to get the litter-level. Because the expiration hasn't passed in this check, it then returns 0 instead of the appropriate value.